Source/DafnyCore/Backends/Rust/Dafny-compiler-rust.dfy

```diff
@@ -3774,17 +3775,25 @@ module {:extern "DCOMP"} DafnyToRustCompiler {
       }
     }
 
-    static method EmitCallToMain(fullName: seq<Name>) returns (s: string) {
-      s := "\nfn main() {\n";
-      var i := 0;
-      while i < |fullName| {
-        if i > 0 {
-          s := s + "::";
-        }
-        s := s + escapeName(fullName[i]);
-        i := i + 1;
+    method EmitCallToMain(companion: Expression, mainMethodName: string, hasArgs: bool) returns (s: string) {
+      s := "\nfn main() {";
```

Comment:
Having test tree vs. formal tree: At this point, the emitted main method is emitted directly into the resulting text file, so any automated refactoring here is useless, as there is zero knowledge of everything that was imported before. We could change the entire architecture but that's too big of a change for this PR.